datafusion/datasource/src/file_scan_config.rs

```diff
@@ -423,10 +403,13 @@ impl FileScanConfigBuilder {
     ///
     /// This method takes ownership of the builder and returns the constructed `FileScanConfig`.
     /// Any unset optional fields will use their default values.
-    pub fn build(self) -> FileScanConfig {
+    ///
+    /// # Errors
+    /// Returns an error if projection pushdown fails or if schema operations fail.
+    pub fn build(self) -> Result<FileScanConfig> {
```

Comment:
Converting this to return Result to properly handle when a FileSource cannot accept any projections. Previously this would have been a silent but that I think was just never hit because all of our FileSources accepted projections